### PR TITLE
Reject cookie authentication

### PR DESCRIPTION
**Pull request by dshanske**
This attempts to reject all requests to the endpoint that might come over cookie authentication, to address more of the Omnibear issue(as the browser based extension is getting confused when someone is logged into the site in the same browser). I tried to simplify this a bit by failing if the auth response is empty.

### CODE REVIEW

File: includes/class-micropub-endpoint.php [hunk 1]
```diff
@@ -90,22 +90,35 @@ public static function register_route() {
 	public static function load_auth() {
 		static::$micropub_auth_response = apply_filters( 'indieauth_response', static::$micropub_auth_response );
 		static::$scopes                 = apply_filters( 'indieauth_scopes', static::$scopes );
+		// Every user should have this capability which reflects the ability to access your user profile and the admin dashboard
+		if ( ! current_user_can( 'read' ) ) {
+			return new WP_Error( 'forbidden', 'Unauthorized', array( 'status' => 403 ) );
+		}
+
+		// If there is no auth response this is cookie authentication which should be rejected
```
Comment (snarfed) [inline includes/class-micropub-endpoint.php]: minor nit: mind explaining why?

**Review by snarfed — APPROVED** [review]
looks great! mind adding a test for this case? then feel free to merge!

File: includes/class-micropub-endpoint.php (hunk 1 above)
Comment (dshanske) [reply includes/class-micropub-endpoint.php]: You shouldn't be able to log into Micropub without a token

Comment (dshanske) [reply includes/class-micropub-endpoint.php]: I probably should use the LOCAL AUTH variable

Comment (snarfed) [reply includes/class-micropub-endpoint.php]: > You shouldn't be able to log into Micropub without a token

right, sorry, i was hoping you could add that to the comment. maybe ideally with a link to the relevant part of the spec. thanks!

Comment (dshanske) [reply includes/class-micropub-endpoint.php]: Was planning on it. But wanted to make sure you got the meaning

**Pull request merged**